cli/azd/pkg/project/framework_service_docker.go

```diff
@@ -619,3 +631,82 @@ func getDockerOptionsWithDefaults(options DockerProjectOptions) DockerProjectOpt
 
 	return options
 }
+
+func addDefaultDockerfileForJava(serviceConfig ServiceConfig) (string, error) {
+	// if Dockerfile not exists, provide a default one
+	if serviceConfig.Language == ServiceLanguageJava {
+		log.Printf("Dockerfile not found for java project %s, will provide a default one", serviceConfig.Name)
+		dockerfileDir, err := os.MkdirTemp("", fmt.Sprintf("%s-%d", serviceConfig.Name, time.Now().Unix()))
+		if err != nil {
+			return "", fmt.Errorf("error creating temp Dockerfile directory: %w", err)
+		}
+		hasParent := serviceConfig.ParentPath != ""
+		dockerfilePath, err := writeDockerfileIntoFs(dockerfileDir, hasParent)
+		if err != nil {
+			return "", err
+		}
+		return dockerfilePath, nil
+	}
+
+	return "", fmt.Errorf("adding default Dockerfile is not supported for language: %s", serviceConfig.Language)
+}
+
+// todo: hardcode jdk-21 as base image here, may need more accurate java version detection.
+const (
+	DockerfileSingleStage = `FROM openjdk:21-jdk-slim
+COPY ./target/*.jar app.jar
+COPY ./target/*.war app.war
+ENTRYPOINT ["sh", "-c", \
+    "if [ -f /app.jar ]; then java -jar /app.jar; \
+    elif [ -f /app.war ]; then java -jar /app.war; \
+    else echo 'No JAR or WAR file found'; fi"]`
+
+	DockerfileMultiStage = `FROM maven:3 AS build
+WORKDIR /app
+COPY . .
+RUN mvn --batch-mode clean package -DskipTests
+
+FROM openjdk:21-jdk-slim
+WORKDIR /
+COPY --from=build /app/target/*.jar app.jar
+COPY --from=build /app/target/*.war app.war
+ENTRYPOINT ["sh", "-c", \
+    "if [ -f /app.jar ]; then java -jar /app.jar; \
+    elif [ -f /app.war ]; then java -jar /app.war; \
+    else echo 'No JAR or WAR file found'; fi"]`
+)
+
+func writeDockerfileIntoFs(path string, hasParent bool) (string, error) {
```

Comment:
Did you test this? will this Dockerfile be used in `azd package` as expected?

Reply:
Yes, tested it. It just like the command: `docker build -t <image name> -f <docker file path> .`